pkg/krmtotf/references.go

```diff
@@ -232,6 +232,16 @@ func resolveTargetFieldValue(r *Resource, tc corekccv1alpha1.TypeConfig) (interf
 }
 
 func resolveDefaultTargetFieldValue(r *Resource, tc corekccv1alpha1.TypeConfig) (interface{}, error) {
+	// Resolve target value from direct resources, when spec.resourceID is unspecified.
+	// Get resourceID from externalRef
+	if r.ResourceConfig.Direct || r.Annotations["alpha.cnrm.cloud.google.com/reconciler"] == "direct" {
```

Comment:
Nit: Can we turn this logic into a single util function in k8s package? They seem to be the same for either TF-based or DCL-based resource.